mysqladmin crash on execution of below command:

  #> mysqladmin -u root -p
  ...
  Segmentation fault (core dumped)

This is probably caused by memory corruption based on working with temp_argv[-1].

RHBZ: #1207041
Upstream report: https://mariadb.atlassian.net/browse/MDEV-7883

diff -up mariadb-5.5.41/client/mysqladmin.cc.bz1207041 mariadb-5.5.41/client/mysqladmin.cc
--- mariadb-5.5.41/client/mysqladmin.cc.bz1207041	2015-03-30 12:37:31.078880914 +0200
+++ mariadb-5.5.41/client/mysqladmin.cc	2015-03-30 13:03:32.442152833 +0200
@@ -1215,7 +1215,8 @@ static char **mask_password(int argc, ch
      }
     argc--;
   }
-  temp_argv[argc]= my_strdup((*argv)[argc], MYF(MY_FAE));
+  if (argc >= 0)
+    temp_argv[argc]= my_strdup((*argv)[argc], MYF(MY_FAE));
   return(temp_argv);
 }
 
